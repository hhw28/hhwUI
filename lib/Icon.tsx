import React, { ReactElement } from "react";

interface IconProps {
  name: String
}

const Icon: React.FunctionComponent<IconProps> = () {
  return <span>icon</span>;
}

export default Icon;
