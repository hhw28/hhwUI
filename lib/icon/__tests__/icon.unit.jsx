import * as renderer from "react-test-renderer";
import React from "react";
import Icon from "../icon";
import { mount } from "enzyme";

describe("Icon", () => {
  it("渲染 render UI success", () => {
    const json = renderer.create(<Icon name="alipay" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  it("事件 onClick", () => {
    const fn = jest.fn();
    const fn1 = jest.fn();
    const component = mount(<Icon name="alipay" onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
