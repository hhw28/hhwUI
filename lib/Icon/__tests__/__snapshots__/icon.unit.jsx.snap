// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 渲染 render UI success 1`] = `
<svg
  className="hhw-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
